File: Data Structures & Algorithms/search-2d-matrix/submission-2.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        System.out.println("Break");
        int high = matrix.length - 1;
        int low = 0;
        int lenOfRow = matrix[0].length - 1;
        while((high - low) >= 1){
            int middle = low + (high - low) / 2;
            if (matrix[middle][0] == target){
                return true;
            }
            else if(target > matrix[middle][0]){
                if (target > matrix[middle][lenOfRow]){
                    low = middle + 1;
                } else {
                    break;
                }
            }
            else if (target < matrix[middle][0]){
                high = middle - 1;
            }
        }
        int row_index = low + (high - low) / 2;
        int result = Arrays.binarySearch(matrix[row_index], target);
        System.out.println(row_index);
        System.out.println(result);
        if (result > -1){
            return true;
        } else {
            return false;
        }
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        high = len(nums) - 1
        low = 0
        while((high - low) >= 0):
            middle = low + ((high - low) // 2)
            if (nums[middle] == target):
                return middle
            elif (target > nums[middle]):
                low = middle + 1
            elif (target < nums[middle]):
                high = middle - 1
        return -1
        